File: typescript/collatz-conjecture/collatz-conjecture.ts
class CollatzConjecture {

    static steps(input: number) {
        if (input <= 0) {
            throw new Error('Only positive numbers are allowed')
        }

        let steps: number = 0

        while (input !== 1) {
            input % 2 === 0 ? input = (input / 2) : (input = input * 3 + 1)
            steps += 1
        }
        return steps
    }
}

export default CollatzConjecture

File: csharp/collatz-conjecture/CollatzConjecture.cs
using System;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        if (number < 1)
        {
            throw new ArgumentOutOfRangeException();
        }

        int steps = 0;

        while (number != 1)
        {
            if (number % 2 == 0)
            {
                number = number / 2;
            }
            else
            {
                number = number * 3 + 1;
            }

            steps += 1;
        }
        return steps;
    }
}
